lib/cached_network_image.dart

```diff
@@ -215,57 +315,94 @@ class CacheObject {
     return null;
   }
 
+  DateTime touched;
+  String url;
+
   Object lock;
   Map _map;
 
-  CacheObject() {
+  CacheObject(String url) {
+    this.url = url;
     _map = new Map();
+    touch();
     lock = new Object();
   }
 
-  CacheObject.fromMap(Map map) {
+  CacheObject.fromMap(String url, Map map) {
+    this.url = url;
     _map = map;
+
+    if (_map.containsKey("touched")) {
+      touched = new DateTime.fromMillisecondsSinceEpoch(_map["touched"]);
+    } else {
+      touch();
+    }
+
     lock = new Object();
   }
 
   Map toMap() {
     return _map;
   }
 
+  touch() {
+    touched = new DateTime.now();
+    _map["touched"] = touched.millisecondsSinceEpoch;
+  }
+
   setDataFromHeaders(Map<String, String> headers) async {
+    //Without a cache-control header we keep the file for a week
+    var ageDuration = new Duration(days: 7);
```

Comment:
interesting. that contradicts earlier Duraton of 30 days (`static Duration maxAgeCacheObject = new Duration(days: 30);`)
I think this is still CacheManager configuration - it belongs to CacheManager

Reply:
I didn't really now how to name this variable, but the functionality is different.
The CacheManager configuration is about when to delete a file and checks when the file is last used by the app.
This 'ageDuration' indicates for how long the file is valid and should normally be set by the cache-control. When the user wants the file but it is not valid anymore it has to pull it from the server again whether the file is still in the cache or not. I didn't want to make it infinite as that might result in problems when a file is changed without the url changing.